unified_planning/model/timing.py

```diff
@@ -28,28 +28,49 @@ class TimepointKind(Enum):
 
 
 class Timepoint:
-    def __init__(self, kind: TimepointKind):
+    def __init__(self, kind: TimepointKind, container=None):
+        """Creates a new timepoint.
+
+        Parameters
+        ----------
+        kind: TimepointKind
+          Kind of the timepoint.
+        container: Optional[Subtask, ActionInstance]
+          Container in which the timepoint is defined.
+          If not set, then a start/end timepoint refers to the enclosing action or method.
+          The container must have an `identifier` property that is a string uniquely identifying it.
+        """
         self._kind = kind
+        self._container = container
 
     def __repr__(self):
         if self._kind == TimepointKind.GLOBAL_START or self._kind == TimepointKind.START:
-            return 'start'
+            qualifier = 'start'
+        else:
+            qualifier = 'end'
+        if self._container is None:
+            return qualifier
         else:
-            return 'end'
+            return f'{qualifier}({self._container.identifier})'
 
     def __eq__(self, oth: object) -> bool:
         if isinstance(oth, Timepoint):
-            return self._kind == oth._kind
+            return self._kind == oth._kind and self._container == oth._container
         else:
             return False
 
     def __hash__(self) -> int:
-        return hash(self._kind)
+        return hash(self._kind) + hash(self._container)
```

Comment:
`ActionInstance` and `Subtask` do not have an `__hash__` method.

Reply:
I added hash/eq for Subtask